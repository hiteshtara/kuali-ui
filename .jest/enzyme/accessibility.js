/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
