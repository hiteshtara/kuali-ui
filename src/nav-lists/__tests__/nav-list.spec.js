/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import { NavList, NavListItem } from '..'

describe('NavList Component', () => {
  it('should render a nav list', () => {
    const wrapper = shallow(
      <NavList>
        <NavListItem iconName='create'>
          Pellentesque habitant morbi tristique senectus
        </NavListItem>
        <NavListItem iconName='description'>Menu Option</NavListItem>
        <NavListItem active iconName='question_answer'>
          Active Menu Option
        </NavListItem>
        <NavListItem iconName='settings'>Configuration</NavListItem>
      </NavList>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with a nested menu', () => {
    const wrapper = shallow(
      <NavList>
        <NavListItem iconName='create'>
          Pellentesque habitant morbi tristique
        </NavListItem>
        <NavListItem active iconName='description'>
          Menu Option
        </NavListItem>
        <NavListItem>Menu Option No Icon</NavListItem>
        <NavListItem iconName='question_answer'>
          Item with Nested Nav
          <NavList>
            <NavListItem>Without Icon</NavListItem>
            <NavListItem iconName='description'>With Icon</NavListItem>
          </NavList>
        </NavListItem>
        <NavListItem iconName='settings'>
          Pellentesque habitant morbi tristique
        </NavListItem>
      </NavList>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
