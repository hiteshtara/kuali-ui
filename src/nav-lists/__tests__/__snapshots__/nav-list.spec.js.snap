// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList Component should render a nav list 1`] = `
<ul
  className="nav-list"
>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="create"
  >
    Pellentesque habitant morbi tristique senectus
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="description"
  >
    Menu Option
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={true}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="question_answer"
  >
    Active Menu Option
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="settings"
  >
    Configuration
  </withInk(NavListItem)>
</ul>
`;

exports[`NavList Component should render with a nested menu 1`] = `
<ul
  className="nav-list"
>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="create"
  >
    Pellentesque habitant morbi tristique
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={true}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="description"
  >
    Menu Option
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
  >
    Menu Option No Icon
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="question_answer"
  >
    Item with Nested Nav
    <NavList
      collapsed={false}
      nested={false}
    >
      <withInk(NavListItem)
        active={false}
        container={[Function]}
        defaultNestedCollapsed={false}
      >
        Without Icon
      </withInk(NavListItem)>
      <withInk(NavListItem)
        active={false}
        container={[Function]}
        defaultNestedCollapsed={false}
        iconName="description"
      >
        With Icon
      </withInk(NavListItem)>
    </NavList>
  </withInk(NavListItem)>
  <withInk(NavListItem)
    active={false}
    container={[Function]}
    defaultNestedCollapsed={false}
    iconName="settings"
  >
    Pellentesque habitant morbi tristique
  </withInk(NavListItem)>
</ul>
`;
